python/tank/api.py

```diff
@@ -592,10 +594,38 @@ def preview_filesystem_structure(self, entity_type, entity_id, engine=None):
                                                       engine)
         return folders
 
+    def get_site_cache_root(self):
+        """
+        Returns the location of the site cache root for this Tank instance.
+        """
+        # Invoke the global method which is generic and can compute any site
+        # cache root.
+        global get_site_cache_root
+        return get_site_cache_root(self.shotgun.base_url)
 
 ##########################################################################################
 # module methods
 
+
+def get_site_cache_root(base_url):
+    """
+    Returns the location of the site cache root based on a site.
+    :param base_url: Site we need to compute the root path for.
+    :returns: An absolute path to the site cache root.
+    """
+    if sys.platform == "darwin":
+        root = os.path.expanduser("~/Library/Caches/Shotgun")
+    elif sys.platform == "win32":
+        root = os.path.join(os.environ["APPDATA"], "Shotgun")
+    elif sys.platform.startswith("linux"):
+        root = os.path.expanduser("~/.shotgun")
```

Comment:
This code was refactored from https://github.com/shotgunsoftware/tk-core/blob/master/hooks/cache_location.py#L92 and now this call site calls the new tank.api method I created.

While I agree that having a cache subfolder would be cleaner, changing that logic at this point would break existing caches. Since they are caches however, they can be rebuilt cleanly over time in the cache subfolder, right?

Shall I update the win32 and linux branch of the”if" to add a cache subfolder?